### PR TITLE
#1: Solution for TwoSum problem

### PR DESCRIPTION
**Pull request by DShydlouski**

### CODE REVIEW

File: ScnSoft-Education/LeetCodeTraining.Tests/TwoSumSolutionTests.cs [hunk 1]
```diff
@@ -0,0 +1,25 @@
...
+            //Assert
+            Assert.Equal(expectedResult.Length, result.Length);
+            Array.Sort(expectedResult);
+            for (int i = 0; i < expectedResult.Length; i++)
+            {
+                Assert.Equal(expectedResult[i], result[i]);
+            }
+        }
+    }
+}
```
[the remark this answers is not in the document]
Comment (DShydlouski) [reply ScnSoft-Education/LeetCodeTraining.Tests/TwoSumSolutionTests.cs:50]: The screenshot has been updated

File: ScnSoft-Education/LeetCodeTraining/TwoSum1/TwoSumSolution.cs [hunk 2]
```diff
@@ -0,0 +1,25 @@
...
+        public static int[] TwoSum(int[] nums, int target)
+        {
+            if (nums == null || nums.Length < 2) return null;
+            for (int i = 0; i < nums.Length - 1; i++)
+            {
+                for (int j = i + 1; j < nums.Length; j++)
+                {
+                    if (nums[i] + nums[j] == target) return new int[] { i, j };
+                }
+            }
```
[the remark this answers is not in the document]
Comment (DShydlouski) [reply ScnSoft-Education/LeetCodeTraining/TwoSum1/TwoSumSolution.cs:20]: Added an extra line

File: ScnSoft-Education/LeetCodeTraining.Tests/TwoSumSolutionTests.cs [hunk 3]
```diff
@@ -0,0 +1,25 @@
...
+    public class TwoSumSolutionTests
+    {
+        [Fact]
+        public void TwoSum_ArrayWith2Nums()
+        {
+            //Arrange
+            int[] nums = { 3, 3 };
+            int target = 6;
+            int[] expectedResult = { 0, 1 };
+            //Act
```
[the remark this answers is not in the document]
Comment (DShydlouski) [reply ScnSoft-Education/LeetCodeTraining.Tests/TwoSumSolutionTests.cs:15]: Added blank lines

File: ScnSoft-Education/LeetCodeTraining/TwoSum1/TwoSumSolution.cs [hunk 4]
```diff
@@ -0,0 +1,25 @@
...
+using System.Threading.Tasks;
+
+namespace LeetCodeTraining.TwoSum1
+{
+    //https://leetcode.com/problems/two-sum/
+    public static class TwoSumSolution
+    {
+        public static int[] TwoSum(int[] nums, int target)
+        {
+            if (nums == null || nums.Length < 2) return null;
```
[the remark this answers is not in the document]
Comment (DShydlouski) [reply ScnSoft-Education/LeetCodeTraining/TwoSum1/TwoSumSolution.cs]: Moved return

File: ScnSoft-Education/LeetCodeTraining/TwoSum1/TwoSumSolution.cs [hunk 5]
```diff
@@ -0,0 +1,25 @@
...
+            for (int i = 0; i < nums.Length - 1; i++)
```
[the remark this answers is not in the document]
Comment (DShydlouski) [reply ScnSoft-Education/LeetCodeTraining/TwoSum1/TwoSumSolution.cs]: Added extra lines

File: ScnSoft-Education/LeetCodeTraining.Tests/TwoSumSolutionTests.cs [hunk 6]
```diff
@@ -0,0 +1,25 @@
...
+            int[] result = TwoSumSolution.TwoSum(nums, target);
+            //Assert
+            Assert.Equal(expectedResult.Length, result.Length);
+            Array.Sort(expectedResult);
```
[the remark this answers is not in the document]
Comment (DShydlouski) [reply ScnSoft-Education/LeetCodeTraining.Tests/TwoSumSolutionTests.cs]: Array.Sort was deleted due to its needlessness

File: ScnSoft-Education/LeetCodeTraining.Tests/TwoSumSolutionTests.cs [hunk 7]
```diff
@@ -0,0 +1,25 @@
...
+            for (int i = 0; i < expectedResult.Length; i++)
```
[the remark this answers is not in the document]
Comment (DShydlouski) [reply ScnSoft-Education/LeetCodeTraining.Tests/TwoSumSolutionTests.cs]: Assertion has been simplified

File: ScnSoft-Education/LeetCodeTraining.Tests/TwoSumSolutionTests.cs [hunk 8]
```diff
@@ -0,0 +1,25 @@
...
+
+namespace LeetCodeTraining.Tests
+{
+    public class TwoSumSolutionTests
+    {
+        [Fact]
+        public void TwoSum_ArrayWith2Nums()
+        {
+            //Arrange
+            int[] nums = { 3, 3 };
```
[the remark this answers is not in the document]
Comment (DShydlouski) [reply ScnSoft-Education/LeetCodeTraining.Tests/TwoSumSolutionTests.cs:11]: Added unit tests for all cases

File: ScnSoft-Education/LeetCodeTraining.Tests/TwoSumSolutionTests.cs [hunk 9]
```diff
@@ -11,15 +11,40 @@ public void TwoSum_ArrayWith2Nums()
...
             int target = 6;
             int[] expectedResult = { 0, 1 };
+
             //Act
             int[] result = TwoSumSolution.TwoSum(nums, target);
+
+            //Assert          
+            Assert.Equal(expectedResult, result);
+        }
+        [Fact]
```
Comment (shomzascnsoft) [inline ScnSoft-Education/LeetCodeTraining.Tests/TwoSumSolutionTests.cs:21]: Please add extra line before [Fact] attribute here and below in the code for better code reading. to separate methods.

File: ScnSoft-Education/LeetCodeTraining/TwoSum1/TwoSumSolution.cs [hunk 10]
```diff
@@ -0,0 +1,25 @@
...
+//https://leetcode.com/problems/two-sum/
+public static class TwoSumSolution
+{
+    public static int[] TwoSum(int[] nums, int target)
+    {
+        if (nums == null || nums.Length < 2)
+        {
+            return null;
+        }
+        for (int i = 0; i < nums.Length - 1; i++)
```
Comment (shomzascnsoft) [inline ScnSoft-Education/LeetCodeTraining/TwoSum1/TwoSumSolution.cs:12]: Please add extra line before this one for better code reading.

**Review by shomzascnsoft — APPROVED** [review]
Please resolve comments with extra lines adding for better code reading and  complete the PR by applying squash merge.

**Review by shomzascnsoft — APPROVED** [review]

**Pull request merged**